support/mock/mockrt.go

```diff
@@ -722,6 +722,9 @@ func (rt *Runtime) ExpectAbort(expected exitcode.ExitCode, f func()) {
 		}
 		// Roll back state change.
 		rt.state = prevState
+
+		// clear expectations unfulfilled due to abort
+		rt.Reset()
```

Comment:
Hmm, I'm a bit torn about this. On the one hand, test code that's expecting an abort can set up just the right expectations for things that happen before the abort point and we should verify them.

On the other hand, a function that factors out some boilerplate might not know that it's expecting an abort, and so set up expectations for success. I'm guessing something like that motivated this here. We could do the reset for those cases in the caller, instead of here.

If it's possible to revert this without much churn, please try. But otherwise ok.

Reply:
I was also a little torn. You correctly identified the problem with the need to consolidate expectation setup in the harness to avoid too much boiler plate. The problem is that we have tests that make make multiple calls to mockrt where the earlier ones expect aborts. In this case, we never verify the expectations, but the expectations unmet due to the abort linger and alter the expectation sequence for the next call.

This scenario is not trivial to debug. I added it to the expect abort call to try to head off this difficulty in the future. If we were going to verify expectations after an abort, I can see how this would be even more confusing. 

I moved the reset up into the caller as you asked. We can work on the pattern for expected aborts from there.